proposals/nnnn-remote-control-onRcStatus-notification.md

```diff
@@ -0,0 +1,92 @@
+
+## Proposed solution
+
+Add an `OnRCStatus` notification RPC to `Mobile_API.xml` and `HMI_API.xml`
+
+SDL shall send OnRCStatus notifications to both the mobile application and the HMI whenever SDL allocates a module to an application or it deallocates a module from an application.
+
+#### Existing data related to the change
+```
+<enum name="ModuleType">
+  <element name="CLIMATE"/>
+  <element name="RADIO"/>
+  <element name="AUDIO"/>
+  <element name="LIGHT">
+  <element name="HMI_SETTINGS">
+  <element name="SEAT">
+</enum>
+```
+
+#### Additions to Mobile_API
+
+```xml
+<element name="OnRCStatusID" value="32784" hexvalue="801A" />
+
+<function name="OnRCStatus" messagetype="notification">
+  <description>Issued by SDL to notify the application about remote control status change on SDL</description>
+  <param name="allocatedModules" type="ModuleType" minsize="0" maxsize="100" array="true" mandatory="true">
+    <description>Contains a list (zero or more) of module types that allocated to the application.</description>
+  </param>
+</function>
+```
+Due to there is no agreement on the ID/ZONE scheme of the resources, we use `ModuleType` here. We shall use the id of a resource if we want to support multiple resources of the same type in the future.
+
+#### Additions to HMI_API
+```xml
+<function name="OnRCStatus" messagetype="notification">
+  <description>Issued by SDL to notify HMI about remote control status change on SDL</description>
+  <param name="appID" type="Integer" mandatory="true">
+    <description>ID of selected application.</description>
+  </param>
+  <param name="allocatedModules" type="ModuleType" minsize="0" maxsize="100" array="true" mandatory="true">
+    <description>Contains a list (zero or more) of module types that allocated to the application.</description>
+  </param>
+</function>
+```
+
+## Potential downsides
+None.
+
+## Impact on existing code
+RPC and Mobile iOS/Android SDK changes:
+- New mobile API parameters support
+
+SDL core changes:
+- Send notifications when a module is allocated/de-allocated to an application.
+
+## Alternatives considered
+In addition to the allocatedModules, we can add an array of free modules in the notification.
+However, that means whenever there is a status change in modules, SDL shall send notifications to all connected RC applications (instead of just sending a notification to the affected application).
+
+```xml<function name="OnRCStatus" messagetype="notification">
+  <description>Issued by SDL to notify the application about remote control status change on SDL</description>
+  <param name="allocatedModules" type="ModuleType" minsize="0" maxsize="100" array="true" mandatory="true">
+    <description>Contains a list (zero or more) of module types that allocated to the application.</description>
```

Comment:
Please change "...module types that allocated to the application." to "...module types that are allocated to the application."

Reply:
updated, thanks